app/src/main/java/io/gnosis/safe/ui/settings/app/AppSettingsFragment.kt

```diff
@@ -67,18 +71,23 @@ class AppSettingsFragment : BaseViewBindingFragment<FragmentSettingsAppBinding>(
 
     private fun setupOwnerKeyView() {
         with(binding) {
-            //TODO: check if app has owner key saved instead of using random
-            if (Random.nextBoolean()) {
+            val address = ownerKeyHandler.retrieveOwnerAddress()
+            val key = ownerKeyHandler.retrieveKey()
+
+            if (address != null && key != BigInteger.ZERO) {
```

Comment:
Should the sentinel value for key also ne null? Or should the sentinel value for address be 0x00?